.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,15 @@
+- id: selene-system
```

Comment:
Rust-based pre-commit does not work because pre-commit hardcodes the `--path` to `.` ([source](https://github.com/pre-commit/pre-commit/blob/14169eb31db6a0ac3e4dfbc38234a8a3e90bc6c1/pre_commit/languages/rust.py#L135)) which leads to build failure with error: `found a virtual manifest at <path>/Cargo.toml instead of a package manifest`. 

Pre-commit maintainer disagrees with making `--path` configurable so it is not possible to add a rust based hook: https://github.com/pre-commit/pre-commit/issues/2931 unless we somehow make it possible to install the binary from root level.

Reply:
I was unsure if it was okay to add a custom install script only for the pre-commit use case. I can add it if that's something we are okay with.

Reply:
I think it would depend on how it would look like. Caching binaries by version may not be trivial and we don't get the same advantages `language: rust` gets, which makes me think it's likely not worth the effort to support this.

Reply:
Agree. That's why went ahead with only setting up `system` and `docker` workflows. Is there some trivial way to make this work by may be a configuration change at our end? I did some preliminary research but did not find much (but then I have not worked with `rust` ever before).

Reply:
Have you looked at how [StyLua handles the pre-commit hook](https://github.com/JohnnyMorganz/StyLua?#pre-commit)?

Reply:
Their `rust` workflow works because they do not use workspaces.

The `python` workflow seems interesting. We can integrate it, but it would depend on a python package (https://pypi.org/project/release-gitter/) to do the release download for us. If that is something we are okay with, I can add it 👍

Reply:
Added GitHub release based pre-commit hook in [5f1a755](https://github.com/Kampfkarren/selene/pull/565/commits/5f1a755b32acc8d36ed96c417eabaf81159018ee)